test/test_transferagent.py

```diff
@@ -316,3 +316,21 @@ def test_handle_metadata_error(self):
             evt = self.transfer_agent.handle_metadata(self.test_site, "foo", "bar")
             assert evt.success is False
             assert isinstance(evt.exception, Exception)
+
+    def test_handle_upload_with_persisted_progress(self):
+        upload_event = UploadEvent(
+            backup_site_name="test_site",
+            file_type=FileType.Basebackup,
+            file_path=Path(self.foo_basebackup_path),
+            source_data=Path(self.foo_basebackup_path),
+            metadata={},
+            file_size=3,
+            callback_queue=CallbackQueue(),
+            remove_after_upload=True
+        )
+
+        self.transfer_agent.handle_upload("test_site", self.foo_basebackup_path, upload_event)
+        updated_progress = PersistedProgress.read(metrics=metrics.Metrics(statsd={}))
```

Comment:
can you patch `PROGRESS_FILE` to point to a tmp_path that way we don't need to do this remove thing

Reply:
ok